fabric-common/lib/BlockDecoder.js

```diff
@@ -700,7 +699,7 @@ function decodePrivateData(privateDataMapProto) {
 			}
 			tx_pvt_read_write_set.ns_pvt_rwset.push(ns_pvt_rwset);
 		}
-		const intIndex = Long.fromValue(txIndex).toInt();
+		const intIndex = fabproto6.util.longFromHash(txIndex).toInt();
 		private_data_map[intIndex] = tx_pvt_read_write_set;
```

Comment:
I think we should use the "txIndex" as is, if it is good enough for the incoming then is should be good enough for the outgoing.

Reply:
as commented bellow, after json stringify it will output to It will output to:
{"\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000":{"value":1},
"\u0001\u0000\u0000\u0000\u0000\u0000\u0000\u0000":{"value":2},
"\u0002\u0000\u0000\u0000\u0000\u0000\u0000\u0000":{"value":3}}

Reply:
In general current fix does not change current interfaces, I have just make it work

I agree that it could have been done better, only now we have an issue on a production, which does not parse.

Would be good to make it work as a first step